### PR TITLE
Armed ships criteria

### PR DESCRIPTION
**Pull request by geoffthemedio**
Changes what counts as an armed ship for combat and blockage purposes.

### CODE REVIEW

**Comment by agrrr3** [comment]
Checked the code; 

The changes are also harmless for FOCS content.

Just Armed::Match in Condition.cpp should remove the reference to HasFighters for consistency:
if (ship->IsArmed() || ship->HasFighters()) --> if (ship->IsArmed())

File: universe/Ship.cpp [hunk 1]
```diff
@@ -244,18 +244,20 @@ bool Ship::IsMonster() const {
 
 bool Ship::IsArmed() const {
     const ShipDesign* design = Design();
-    if (design)
-        return design->IsArmed();
-    else
+    if (!design || !design->IsArmed())
```
[the remark this answers is not in the document]
Comment (geoffthemedio) [reply universe/Ship.cpp]: fixed?

**Comment by geoffthemedio** [comment]
Reworked Ship:: and ShipDesign::IsArmed and separated out ShipDesign::HasDirectWeapons

File: universe/Ship.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (Morlic-fo) [reply universe/Ship.cpp]: I think so.

**Pull request merged**